torch/csrc/utils/object_ptr.h

```diff
@@ -5,9 +5,9 @@
 template<class T>
 class THPPointer {
 public:
-  THPPointer(): ptr(nullptr) {};
-  THPPointer(T *ptr): ptr(ptr) {};
-  THPPointer(THPPointer &&p) { free(); ptr = p.ptr; p.ptr = nullptr; };
+  explicit THPPointer(): ptr(nullptr) {};
```

Comment:
I suspect we should probably follow the convention of std::unique_ptr (http://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr) and only have line 9 be marked explicit.